core/types/block.libevm_test.go

```diff
@@ -94,19 +132,31 @@ func TestHeaderHooks(t *testing.T) {
 		require.NoErrorf(t, err, "rlp.DecodeBytes(%#x)", input)
 
 		assert.Equal(t, input, stub.gotRawRLPToDecode, "raw RLP received by hooks")
-		assert.Equalf(t, &stub.setHeaderToOnDecode, hdr, "%T after RLP decoding with hook", hdr)
+		assert.Equalf(t, &stub.setHeaderToOnUnmarshalOrDecode, hdr, "%T after RLP decoding with hook", hdr)
 	})
 
 	t.Run("error_propagation", func(t *testing.T) {
+		errMarshal := errors.New("whoops")
+		errUnmarshal := errors.New("is it broken?")
 		errEncode := errors.New("uh oh")
 		errDecode := errors.New("something bad happened")
 
 		hdr := new(Header)
-		extras.Header.Set(hdr, &stubHeaderHooks{
-			errEncode: errEncode,
-			errDecode: errDecode,
-		})
+		setStub := func() {
+			extras.Header.Set(hdr, &stubHeaderHooks{
+				errMarshal:   errMarshal,
+				errUnmarshal: errUnmarshal,
+				errEncode:    errEncode,
+				errDecode:    errDecode,
+			})
+		}
+
+		setStub()
+		_, err := json.Marshal(hdr)
+		assert.ErrorIs(t, err, errMarshal, "via json.Marshal()") //nolint:testifylint // require is inappropriate here as we wish to keep going
+		assert.Equal(t, errUnmarshal, json.Unmarshal([]byte("{}"), hdr), "via json.Unmarshal()")
```

Comment:
nit

```suggestion
		err = json.Unmarshal([]byte("{}")
		assert.Equal(t, errUnmarshal, err, hdr), "via json.Unmarshal()")
```

Reply:
Done in d1f481661 to include the other tests too.